### PR TITLE
Moving to @react-native-community/eslint-config

### PR DESCRIPTION
**Pull request by mmazzarolo**
A few alignment to the community standards:
- [x] Moved from `yarn` to `npm`  
- [x] Moved the `.ts` definitions to the new `typings` dir
- [x] Moved to `@react-native-community/eslint-config` (and ran it with the `--fix` option on the `src` dir)

### CODE REVIEW

**Comment by matt-oakes** [comment]
I've only take a quick look and it looks good!

One small issue is that the prettier config is missing from the `.prettierrc` file in the project root. It should look like this to match the React Native core settings:

https://github.com/react-native-community/react-native-netinfo/blob/matt-oakes/new-api/.prettierrc

When you run lint fix again, you should notice that there is no longer spaces around the `{}`'s in imports (🤢!)

**Comment by rewieer** [comment]
Looks good to me, just added a little comment

**Comment by mmazzarolo** [comment]
Thanks for the reviews!

> ooks good to me, just added a little comment
I don't see it 🤔 
  
> One small issue is that the prettier config is missing from the .prettierrc file in the project root. It should look like this to match the React Native core settings:
  
Wait, is it a community standard? 👀 
I checked a few community repos and they were all using different `.prettierrc` files...
Also, wouldn't a root `.prettierrc` cause conflicts with the [config itself](https://github.com/facebook/react-native/blob/master/packages/eslint-config-react-native-community/index.js) (prettier rules can be defined there as well).

File: src/utils.js [hunk 1]
```diff
@@ -1,30 +1,30 @@
-import { Dimensions } from "react-native";
-import * as animatable from "react-native-animatable";
+import { Dimensions } from 'react-native';
+import * as animatable from 'react-native-animatable';
 
-const { height, width } = Dimensions.get("window");
+const { height, width } = Dimensions.get('window');
 
 // Since react-native-animatable applies by default a margin of 100 to its
 // sliding animation, we reset them here overriding the margin to 0.
```
[the remark this answers is not in the document]
Comment (rewieer) [reply src/utils.js]: @mmazzarolo here it is

[the remark this answers is not in the document]
Comment (mmazzarolo) [reply src/utils.js]: The comment is still valid but you're right, it doesn't make too much sense in that position. I moved it where we set the margins to `0` 👍

**Comment by mmazzarolo** [comment]
I'm merging it, @matt-oakes let me know what do you think about my `.prettierrc` comment above ⬆️ (eventually I'll add it in a new PR)

**Comment by matt-oakes** [comment]
@mmazzarolo Having the Prettier config in the ESLint config was causing issues and was never a recommended way to do it (code editors can't use the ESLint prettier settings, but can used the shared `.prettierrc` one). It's since been removed. See here: https://github.com/facebook/react-native/pull/24635

**Comment by mmazzarolo** [comment]
@matt-oakes yeah, that was confusing to me. Thanks for taking care of it 🙌 
Will add the `.prettierrc` in a next PR.

**Pull request merged**